### PR TITLE
Secure sandbox endpoint url and pagination.

### PR DESCRIPTION
**Pull request by slvsridhar**
Included secure sandbox endpoint url and pagination changes for Observation, CarePlan and Diagnostic Report resource

### CODE REVIEW

File: lib/resources/example_json/soarian_examples_link_extension.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+module Cerner
```
Comment (FFSol) [inline lib/resources/example_json/soarian_examples_link_extension.rb]:
```suggestion
# frozen_string_literal: true

module Cerner
  module Resources
    SOARIAN_LINK_EXTENSION ||= {
      "type": "searchset",
      "link": [
        {
          "relation": "self",
          "url": "http://fhir-myrecord.sandboxcerner.com/dstu2/Observation?patientId=2A39DA0B-61D8-4F95-BE00-9CA30EE01819&_format=json&page=2"
        },
        {
          "relation": "next",
          "url": "http://fhir-myrecord.sandboxcerner.com/dstu2/Observation?patientId=2A39DA0B-61D8-4F95-BE00-9CA30EE01819&_format=json&page=3"
        }
      ]
    }
  end
end
```

File: lib/resources/example_json/soarian_examples_observation.rb [hunk 2]
```diff
@@ -11,7 +11,11 @@ module Resources
       "link": [
         {
           "relation": "self",
-          "url": "base-url/123abc/Observation?patientId=D4292B7B-AEDF-4CEF-B783-BD3AA1B0DD27&_format=json"
+          "url": "base-url/123abc/Observation?patientId=D4292B7B-AEDF-4CEF-B783-BD3AA1B0DD27&_format=json&page=2"
+        },
+		{
```
Comment (FFSol) [inline lib/resources/example_json/soarian_examples_observation.rb]: Could you line up these curly braces

File: lib/resources/example_json/soarian_examples_diagnostic_report.rb [hunk 3]
```diff
@@ -11,7 +11,11 @@ module Resources
       "link": [
         {
           "relation": "self",
-          "url": "base-url/123abc/DiagnosticReport?patientId=D4292B7B-AEDF-4CEF-B783-BD3AA1B0DD27&_format=json"
+          "url": "base-url/123abc/DiagnosticReport?patientId=D4292B7B-AEDF-4CEF-B783-BD3AA1B0DD27&_format=json&page=2"
+        },
+		{
```
Comment (FFSol) [inline lib/resources/example_json/soarian_examples_diagnostic_report.rb]: nit: formatting

File: lib/resources/example_json/soarian_examples_care_plan.rb [hunk 4]
```diff
@@ -33,7 +33,11 @@ module Resources
       "link": [
         {
           "relation": "self",
-          "url": "base-url/123abc/CarePlan?patientId=D4292B7B-AEDF-4CEF-B783-BD3AA1B0DD27&_format=json"
+          "url": "base-url/123abc/CarePlan?patientId=D4292B7B-AEDF-4CEF-B783-BD3AA1B0DD27&_format=json&page=2"
+        },
+		{
```
Comment (FFSol) [inline lib/resources/example_json/soarian_examples_care_plan.rb]: nit: formatting

File: content/soarian/dstu2.md [hunk 5]
```diff
@@ -53,6 +53,15 @@ For example:
 Where `https://fhir-myrecord.sandboxcerner.com/dstu2/abcd1234/` is the the `:serviceRootURL`, `Condition` is the 
 `:resource`, and `patientId=4342008` is the `:parameters`.
 
+### Secure Sandbox
+
+The secure sandbox instance can be used for testing an application with [authorization](#authorization). The service root URL for the sandbox instance is:
+
+`https://fhir-myrecord.sandboxcerner.com/dstu2/0e885770-571b-4c0c-b30f-21df9a058d0d`
+
+Example: `https://fhir-myrecord.sandboxcerner.com/dstu2/0e885770-571b-4c0c-b30f-21df9a058d0d/Condition?patientId= D4292B7B-AEDF-4CEF-B783-BD3AA1B0DD27`
```
Comment (MaxPhilips) [inline content/soarian/dstu2.md]: there's a space in this URL

File: content/soarian/dstu2.md [hunk 6]
```diff
@@ -53,6 +53,15 @@ For example:
 Where `https://fhir-myrecord.sandboxcerner.com/dstu2/abcd1234/` is the the `:serviceRootURL`, `Condition` is the 
 `:resource`, and `patientId=4342008` is the `:parameters`.
 
+### Secure Sandbox
+
+The secure sandbox instance can be used for testing an application with [authorization](#authorization). The service root URL for the sandbox instance is:
+
+`https://fhir-myrecord.sandboxcerner.com/dstu2/0e885770-571b-4c0c-b30f-21df9a058d0d`
```
Comment (MaxPhilips) [inline content/soarian/dstu2.md]: this looks like the same URL as above - does Soarian have an open and a secure sandbox? seems that there's no open, so I don't think the h3 for Secure Sandbox makes sense.

File: lib/resources/example_json/soarian_examples_link_extension.rb (hunk 1 above)
Comment (MaxPhilips) [reply lib/resources/example_json/soarian_examples_link_extension.rb]: not needed

File: lib/resources/example_json/soarian_examples_link_extension.rb [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+  module Resources
+    SOARIAN_LINK_EXTENSION ||= {
+"type": "searchset",
```
Comment (MaxPhilips) [inline lib/resources/example_json/soarian_examples_link_extension.rb]: format with two space indentation

File: lib/resources/example_json/soarian_examples_observation.rb [hunk 8]
```diff
@@ -11,7 +11,11 @@ module Resources
       "link": [
         {
           "relation": "self",
-          "url": "base-url/123abc/Observation?patientId=D4292B7B-AEDF-4CEF-B783-BD3AA1B0DD27&_format=json"
+          "url": "base-url/123abc/Observation?patientId=D4292B7B-AEDF-4CEF-B783-BD3AA1B0DD27&_format=json&page=2"
```
Comment (MaxPhilips) [inline lib/resources/example_json/soarian_examples_observation.rb]: can we change 'base-url' now?

File: content/soarian/dstu2.md [hunk 9]
```diff
@@ -113,6 +122,20 @@ ID                                | Value\[x] Type      | Description
...
+The pagination links are included in the Bundle when a resource returns multiple items. It is important to follow these Link header values instead of constructing your own URLs.
+
+<%= json(:SOARIAN_LINK_EXTENSION) %>
+
+The possible `relation` values are:
+
+|Name     | Description
+|---------|----------------------------------------
+|`self`   | Shows the URL of the current page of results.
+|`next`   | Shows the URL of the immediate next page of results.
```
Comment (MaxPhilips) [inline content/soarian/dstu2.md]: prefer to format tables this way:

```
 Column | Column
--------|-------------
 data   | data
 data   | longer data

* no leading or trailing pipes
* column width set to widest row
* dashes cover entire table width

Comment (MaxPhilips) [reply content/soarian/dstu2.md]: we'll need to note that 'previous' links can return. seems like this was missed for Millennium - I would prefer to address that as well with this change

File: lib/resources/example_json/soarian_examples_observation.rb (hunk 8 above)
Comment (daliboz) [reply lib/resources/example_json/soarian_examples_observation.rb]: +1 - these should use examples from the sandbox (that actually work) now that it's been set up for Clinicals

File: lib/resources/example_json/soarian_examples_link_extension.rb (hunk 1 above)
Comment (slvsridhar) [reply lib/resources/example_json/soarian_examples_link_extension.rb]: "module Cerner" is the pattern followed in all the files and we are adhering to that.

File: lib/resources/example_json/soarian_examples_observation.rb (hunk 2 above)
Comment (slvsridhar) [reply lib/resources/example_json/soarian_examples_observation.rb]: Formatting has been taken care.

File: lib/resources/example_json/soarian_examples_diagnostic_report.rb (hunk 3 above)
Comment (slvsridhar) [reply lib/resources/example_json/soarian_examples_diagnostic_report.rb]: Formatting has been taken care

File: lib/resources/example_json/soarian_examples_care_plan.rb (hunk 4 above)
Comment (slvsridhar) [reply lib/resources/example_json/soarian_examples_care_plan.rb]: Formatting has been taken care

File: content/soarian/dstu2.md (hunk 5 above)
Comment (slvsridhar) [reply content/soarian/dstu2.md]: Removed space in the URL

File: content/soarian/dstu2.md (hunk 6 above)
Comment (slvsridhar) [reply content/soarian/dstu2.md]: 1.The First Link under secure sandbox section indicates the service root url for secure sandbox. This follows the general format specified under the “Service root URL” section which is why we decided to hyperlink the aforementioned section under “secure sandbox”
2.The Second link indicates an example of the URL for a given recourse call.
Therefore the 2 links present under secure sandbox need to be present.

File: lib/resources/example_json/soarian_examples_link_extension.rb (hunk 7 above)
Comment (slvsridhar) [reply lib/resources/example_json/soarian_examples_link_extension.rb]: Formatting has been taken care

File: lib/resources/example_json/soarian_examples_observation.rb (hunk 8 above)
Comment (slvsridhar) [reply lib/resources/example_json/soarian_examples_observation.rb]: Yes. We have updated the base url with the actual secure sandbox url.

File: content/soarian/dstu2.md (hunk 9 above)
Comment (slvsridhar) [reply content/soarian/dstu2.md]: Table formatting has been done.  Soarian does not support 'previous' link and that is why it was not mentioned.

**Comment by slvsridhar** [comment]
Hi @MaxPhilips , @FFSol and @daliboz - I have worked on the comments and updated. Could you please review and share your thoughts?

File: lib/resources/example_json/soarian_examples_observation.rb (hunk 8 above)
Comment (daliboz) [reply lib/resources/example_json/soarian_examples_observation.rb]: @slvsridhar  I'm not seeing any updates in this patch/PR. Have you pushed those changes to the branch the PR is from?

File: content/soarian/dstu2.md [hunk 10]
```diff
@@ -113,6 +122,20 @@ ID                                | Value\[x] Type      | Description
 <%= json(:SOARIAN_CUSTOM_EXTENSION) %>
 
 
+## Pagination
+
+The pagination links are included in the Bundle when a resource returns multiple items. It is important to follow these Link header values instead of constructing your own URLs.
+
+<%= json(:SOARIAN_LINK_EXTENSION) %>
```
Comment (daliboz) [inline content/soarian/dstu2.md]: Is this really an extension or is it using the standard links in the bundle?

Comment (daliboz) [reply content/soarian/dstu2.md]: (in examples, it looks normal, so the naming of this with "extension" is confusing - could also just link to the FHIR doc that discusses what these links mean)

Comment (Anirudh-Kumar) [reply content/soarian/dstu2.md]: You're right. i think naming them "SOARIAN_FHIR_LINKS"  would make more sense. Any suggestions for other nomenclature is welcome.

File: lib/resources/example_json/soarian_examples_link_extension.rb (hunk 1 above)
Comment (slvsridhar) [reply lib/resources/example_json/soarian_examples_link_extension.rb]: Changes are present here. https://github.com/cerner/fhir.cerner.com/pull/232/commits/416d911dabbb8e9fdd2602dbf306ae20ee43e10d

File: lib/resources/example_json/soarian_examples_observation.rb (hunk 2 above)
Comment (slvsridhar) [reply lib/resources/example_json/soarian_examples_observation.rb]: Changes are present here. https://github.com/cerner/fhir.cerner.com/pull/232/commits/416d911dabbb8e9fdd2602dbf306ae20ee43e10d

File: lib/resources/example_json/soarian_examples_diagnostic_report.rb (hunk 3 above)
Comment (slvsridhar) [reply lib/resources/example_json/soarian_examples_diagnostic_report.rb]: Changes are present here. https://github.com/cerner/fhir.cerner.com/pull/232/commits/416d911dabbb8e9fdd2602dbf306ae20ee43e10d

File: lib/resources/example_json/soarian_examples_care_plan.rb (hunk 4 above)
Comment (slvsridhar) [reply lib/resources/example_json/soarian_examples_care_plan.rb]: Changes are present here. https://github.com/cerner/fhir.cerner.com/pull/232/commits/416d911dabbb8e9fdd2602dbf306ae20ee43e10d

File: content/soarian/dstu2.md (hunk 5 above)
Comment (slvsridhar) [reply content/soarian/dstu2.md]: Changes are present here. https://github.com/cerner/fhir.cerner.com/pull/232/commits/416d911dabbb8e9fdd2602dbf306ae20ee43e10d

File: content/soarian/dstu2.md (hunk 6 above)
Comment (slvsridhar) [reply content/soarian/dstu2.md]: Changes are present here. https://github.com/cerner/fhir.cerner.com/pull/232/commits/416d911dabbb8e9fdd2602dbf306ae20ee43e10d

File: lib/resources/example_json/soarian_examples_link_extension.rb (hunk 7 above)
Comment (slvsridhar) [reply lib/resources/example_json/soarian_examples_link_extension.rb]: Changes are present here. https://github.com/cerner/fhir.cerner.com/pull/232/commits/416d911dabbb8e9fdd2602dbf306ae20ee43e10d

File: lib/resources/example_json/soarian_examples_observation.rb (hunk 8 above)
Comment (slvsridhar) [reply lib/resources/example_json/soarian_examples_observation.rb]: @daliboz  yes. I have pushed the changes in the same branch and Changes are present here. https://github.com/cerner/fhir.cerner.com/pull/232/commits/416d911dabbb8e9fdd2602dbf306ae20ee43e10d

File: content/soarian/dstu2.md (hunk 9 above)
Comment (slvsridhar) [reply content/soarian/dstu2.md]: Changes are present here. https://github.com/cerner/fhir.cerner.com/pull/232/commits/416d911dabbb8e9fdd2602dbf306ae20ee43e10d

File: content/soarian/dstu2.md (hunk 10 above)
Comment (slvsridhar) [reply content/soarian/dstu2.md]: Modified link example URL and changes are present here https://github.com/cerner/fhir.cerner.com/pull/232/commits/dbec31a2c6140761e3a20ba9d91a4b8858c141bc

**Pull request merged**